GANDLF/entrypoints/run.py

```diff
@@ -0,0 +1,291 @@
+#!usr/bin/env python
+# -*- coding: utf-8 -*-
+import logging
+import os
+import argparse
+import ast
+import sys
+import traceback
+from typing import Optional
+
+from deprecated import deprecated
+import click
+
+from GANDLF import version
+from GANDLF.cli import main_run, copyrightMessage
+from GANDLF.entrypoints import append_copyright_to_help
+from GANDLF.utils import logger_setup
+
+
+def _run(
+    config: str,
+    input_data: str,
+    train_flag: bool,
+    model_dir: str,
+    device: str,
+    reset_flag: bool,
+    resume_flag: bool,
+    output_path: Optional[str],
+):
+    if model_dir is None and output_path:
+        model_dir = output_path
+
+    assert model_dir is not None, "Missing required parameter: model_dir"
+
+    if os.path.isdir(input_data):
+        # Is this a fine assumption to make?
+        # Medperf models receive the data generated by the data preparator mlcube
+        # We can therefore ensure the output of that mlcube contains a data.csv file
+        filename = "data.csv"
+        input_data = os.path.join(input_data, filename)
+
+    if not train_flag:
+        # TODO: print a warning if any of these flags is activated. They are not available for inference mode
+        #  Maybe user misconfigured the command.
+        # if inference mode, then no need to check for reset/resume
+        reset_flag, resume_flag = False, False
+
+    if reset_flag and resume_flag:
+        logging.warning(
+            "'reset' and 'resume' are mutually exclusive; 'resume' will be used."
+        )
+        reset_flag = False
+
+    # TODO: check that output_path is not passed in training mode;
+    #  maybe user misconfigured the command
+
+    logging.debug(f"{config=}")
+    logging.debug(f"{input_data=}")
+    logging.debug(f"{train_flag=}")
+    logging.debug(f"{model_dir=}")
+    logging.debug(f"{device=}")
+    logging.debug(f"{reset_flag=}")
+    logging.debug(f"{resume_flag=}")
+    logging.debug(f"{output_path=}")
+
+    try:
+        main_run(
+            data_csv=input_data,
+            config_file=config,
+            model_dir=model_dir,
+            train_mode=train_flag,
+            device=device,
+            resume=resume_flag,
+            reset=reset_flag,
+            output_dir=output_path,
+        )
+    except Exception:
+        # TODO: why so? Why not just default way when exception is printed
+        #  and process exits with code 1 automatically?
+        sys.exit("ERROR: " + traceback.format_exc())
```

Comment:
As now GaNDLF uses logging, maybe here it would be also worth to log the Error and traceback before exit